mgmt_cli_test.py

```diff
@@ -602,6 +603,53 @@ def test_basic_backup(self, ks_names: list = None):
         mgr_cluster.delete()  # remove cluster at the end of the test
         self.log.info('finishing test_basic_backup')
 
+    def test_no_delta_backup_at_disabled_compaction(self):
+        """The purpose of test is to check that delta backup (no changes to DB between backups) takes time -> 0.
+
+        Important test precondition is to disable compaction on all nodes in the cluster.
+        Otherwise, new set of SSTables is created what ends up in the situation that almost no deduplication is applied.
+
+        For more details https://github.com/scylladb/scylla-manager/issues/3936#issuecomment-2277611709
+        """
+        self.log.info('starting test_consecutive_backups')
+
+        self.log.info('Run write stress')
+        self.run_prepare_write_cmd()
+
+        self.log.info('Disable compaction for every node in the cluster')
+        compaction_ops = CompactionOps(cluster=self.db_cluster)
+        for node in self.db_cluster.nodes:
+            compaction_ops.disable_autocompaction_on_ks_cf(node=node)
+
+        self.log.info('Prepare Manager')
+        manager_tool = mgmt.get_scylla_manager_tool(manager_node=self.monitors.nodes[0])
+        mgr_cluster = self._ensure_and_get_cluster(manager_tool, force_add=True)
+
+        self.log.info('Run backup #1')
+        backup_task_1 = mgr_cluster.create_backup_task(location_list=self.locations)
+        backup_task_1_status = backup_task_1.wait_and_get_final_status(timeout=3600)
+        assert backup_task_1_status == TaskStatus.DONE, \
+            f"Backup task ended in {backup_task_1_status} instead of {TaskStatus.DONE}"
+        self.log.info(f'Backup task #1 duration - {backup_task_1.duration}')
+
+        self.log.info('Run backup #2')
+        backup_task_2 = mgr_cluster.create_backup_task(location_list=self.locations)
+        backup_task_2_status = backup_task_2.wait_and_get_final_status(timeout=60)
+        assert backup_task_2_status == TaskStatus.DONE, \
+            f"Backup task ended in {backup_task_2_status} instead of {TaskStatus.DONE}"
+        self.log.info(f'Backup task #2 duration - {backup_task_2.duration}')
+
+        assert backup_task_2.duration < timedelta(seconds=15), "No-delta backup took more than 15 seconds"
+
+        self.log.info('Verify restore from backup #2')
+        self.verify_backup_success(mgr_cluster=mgr_cluster, backup_task=backup_task_2,
```

Comment:
maybe worth to delete backup 1 to see if it doesn't break backup 2?

Reply:
IMO It might be a good improvement for the test.
@Michal-Leszczynski What do you think about it?

Reply:
I don't fully understand what "delete backup 1" means.
Is it about deleting the task? Or is it about deleting the snapshot?
We could do the first one, although this wouldn't be relevant to the test.
We can't do the second one, as then the backup would need to be fully re-uploaded.

Reply:
If we can't do the second one, then it actually doesn't make sense.

Reply:
ok, so it means delta-backup strongly relies on previous snapshots and these can't be deleted. I hope it's documented somewhere.

Reply:
Hm, just curious, in `backup delete` [doc](https://manager.docs.scylladb.com/stable/sctool/backup.html#backup-delete) there is such info:
```
Deduplicated files are persisted unless their reference count drops to zero.
```
I suppose the flow with running backup deletion with `sctool backup delete --snapshot-tag BACKUP_1_TAG` should keep deduplicated files required for backup 2 and we are kinda safe to restore from it.
@Michal-Leszczynski What do you think?

Reply:
Whatever, guys @soyacz @Michal-Leszczynski, if you don't mind I'd suggest to move this investigation and possible test improvement to a separate ticket to not block this PR now since we don't have much time at the moment to work on it?